### ISSUE TITLE
off line install

### ISSUE THREAD

**Issue by crazy-canux** [hexojs/hexo#1823]
1. download the package.
2.unzip the package to  /home/user/Download/hexo-master
3.sudo hexo install  /home/user/Download/hexo-master

still get the error.
Is there any way to install the package offline?

**Comment by samwhelp** [comment]
you should read manual first!
figure out  how to install node package by npm.

https://docs.npmjs.com/cli/npm

```
global mode:
local mode:
```

https://docs.npmjs.com/cli/install

```
npm install <folder>:
npm install <tarball file>:
```

step 1: install [hexo-cli](https://github.com/hexojs/hexo-cli) into global


but some hexo-cli [dependencies](https://github.com/hexojs/hexo-cli/blob/master/package.json) should install first.


step 2: install [hexo](https://github.com/hexojs/hexo) into local

but some hexo [dependencies](https://github.com/hexojs/hexo/blob/master/package.json) should install first.

PS: it's not easy.

**Comment by crazy-canux** [comment]
Ok, I read the docs.
But still get  the error:

0 info it worked if it ends with ok
1 verbose cli [ '/usr/local/bin/node', '/usr/local/bin/npm', 'install', '-g' ]
2 info using npm@2.14.20
3 info using node@v4.4.0
4 verbose install initial load of /usr/local/lib/package.json
5 verbose readDependencies loading dependencies from /usr/local/lib/package.json
6 silly cache add args [ '.', null ]
7 verbose cache add spec .
8 silly cache add parsed spec Result {
8 silly cache add   raw: '.',
8 silly cache add   scope: null,
8 silly cache add   name: null,
8 silly cache add   rawSpec: '.',
8 silly cache add   spec: '/home/chengca/Downloads/hexo-cli-master',
8 silly cache add   type: 'directory' }
9 verbose addLocalDirectory /home/chengca/.npm/hexo-cli/1.0.1/package.tgz not in flight; packing
10 verbose correctMkdir /home/chengca/.npm correctMkdir not in flight; initializing
11 verbose tar pack [ '/home/chengca/.npm/hexo-cli/1.0.1/package.tgz',
11 verbose tar pack   '/home/chengca/Downloads/hexo-cli-master' ]
12 verbose tarball /home/chengca/.npm/hexo-cli/1.0.1/package.tgz
13 verbose folder /home/chengca/Downloads/hexo-cli-master
14 info prepublish hexo-cli@1.0.1
15 verbose addLocalTarball adding from inside cache /home/chengca/.npm/hexo-cli/1.0.1/package.tgz
16 verbose correctMkdir /home/chengca/.npm correctMkdir not in flight; initializing
17 silly cache afterAdd hexo-cli@1.0.1
18 verbose afterAdd /home/chengca/.npm/hexo-cli/1.0.1/package/package.json not in flight; writing
19 verbose correctMkdir /home/chengca/.npm correctMkdir not in flight; initializing
20 verbose afterAdd /home/chengca/.npm/hexo-cli/1.0.1/package/package.json written
21 silly install resolved [ { name: 'hexo-cli',
21 silly install resolved     version: '1.0.1',
21 silly install resolved     description: 'Command line interface for Hexo',
21 silly install resolved     main: 'lib/hexo',
21 silly install resolved     bin: { hexo: './bin/hexo' },
21 silly install resolved     scripts:
21 silly install resolved      { eslint: 'eslint .',
21 silly install resolved        jscs: 'jscs .',
21 silly install resolved        test: 'mocha test/index.js',
21 silly install resolved        'test-cov': 'istanbul cover --print both _mocha -- test/index.js' },
21 silly install resolved     directories: { lib: './lib', bin: './bin' },
21 silly install resolved     repository:
21 silly install resolved      { type: 'git',
21 silly install resolved        url: 'git+https://github.com/hexojs/hexo-cli.git' },
21 silly install resolved     homepage: 'https://hexo.io/',
21 silly install resolved     keywords: [ 'website', 'blog', 'cms', 'framework', 'hexo', 'cli' ],
21 silly install resolved     author:
21 silly install resolved      { name: 'Tommy Chen',
21 silly install resolved        email: 'tommy351@gmail.com',
21 silly install resolved        url: 'http://zespia.tw' },
21 silly install resolved     license: 'MIT',
21 silly install resolved     dependencies:
21 silly install resolved      { abbrev: '^1.0.7',
21 silly install resolved        bluebird: '^3.3.0',
21 silly install resolved        chalk: '^1.1.1',
21 silly install resolved        'hexo-fs': '^0.1.5',
21 silly install resolved        'hexo-log': '^0.1.1',
21 silly install resolved        'hexo-util': '^0.5.3',
21 silly install resolved        minimist: '^1.2.0',
21 silly install resolved        'object-assign': '^4.0.1',
21 silly install resolved        tildify: '^1.1.2' },
21 silly install resolved     devDependencies:
21 silly install resolved      { chai: '^3.5.0',
21 silly install resolved        'chai-as-promised': '^5.2.0',
21 silly install resolved        eslint: '^1.10.3',
21 silly install resolved        'eslint-config-hexo': '^1.0.2',
21 silly install resolved        'hexo-renderer-marked': '^0.2.9',
21 silly install resolved        istanbul: '^0.4.2',
21 silly install resolved        jscs: '^2.9.0',
21 silly install resolved        'jscs-preset-hexo': '^1.0.1',
21 silly install resolved        mocha: '^2.4.5',
21 silly install resolved        proxyquire: '^1.7.4',
21 silly install resolved        rewire: '^2.5.1',
21 silly install resolved        sinon: '^1.17.3' },
21 silly install resolved     readme: '# hexo-cli\n\n[![Build Status](https://travis-ci.org/hexojs/hexo-cli.svg?branch=master)](https://travis-ci.org/hexojs/hexo-cli)  [![NPM version](https://badge.fury.io/js/hexo-cli.svg)](http://badge.fury.io/js/hexo-cli) [![Coverage Status](https://coveralls.io/repos/hexojs/hexo-cli/badge.svg?branch=master)](https://coveralls.io/r/hexojs/hexo-cli?branch=master) [![Build status](https://ci.appveyor.com/api/projects/status/aa44wp75752x1t47/branch/master?svg=true)](https://ci.appveyor.com/project/tommy351/hexo-cli/branch/master)\n\nCommand line interface for Hexo.\n\n## Installation\n\n``` bash\n$ npm install hexo-cli -g\n```\n\n## License\n\nMIT',
21 silly install resolved     readmeFilename: 'README.md',
21 silly install resolved     bugs: { url: 'https://github.com/hexojs/hexo-cli/issues' },
21 silly install resolved     _id: 'hexo-cli@1.0.1',
21 silly install resolved     _shasum: '8e4f5b4cb303da82feba044eb76f4c4f5c29931a',
21 silly install resolved     _from: '../../home/chengca/Downloads/hexo-cli-master',
21 silly install resolved     _resolved: 'file:../../home/chengca/Downloads/hexo-cli-master' } ]
22 info install hexo-cli@1.0.1 into /usr/local/lib
23 info installOne hexo-cli@1.0.1
24 verbose installOne of hexo-cli to /usr/local/lib not in flight; installing
25 verbose correctMkdir /home/chengca/.npm/_locks correctMkdir not in flight; initializing
26 verbose makeDirectory /home/chengca/.npm/_locks creation not in flight; initializing
27 silly makeDirectory /home/chengca/.npm/_locks uid: 1002 gid: 1003
28 verbose lock using /home/chengca/.npm/_locks/hexo-cli-d89d95637eeaa0d8.lock for /usr/local/lib/node_modules/hexo-cli
29 silly install write writing hexo-cli 1.0.1 to /usr/local/lib/node_modules/hexo-cli
30 verbose unbuild lib/node_modules/hexo-cli
31 silly gentlyRm /usr/local/lib/node_modules/hexo-cli is being purged from base /usr/local
32 verbose gentlyRm don't care about contents; nuking /usr/local/lib/node_modules/hexo-cli
33 verbose tar unpack /home/chengca/.npm/hexo-cli/1.0.1/package.tgz
34 verbose tar unpacking to /usr/local/lib/node_modules/hexo-cli
35 silly gentlyRm /usr/local/lib/node_modules/hexo-cli is being purged
36 verbose gentlyRm don't care about contents; nuking /usr/local/lib/node_modules/hexo-cli
37 silly gunzTarPerm modes [ '755', '644' ]
38 silly gunzTarPerm extractEntry package.json
39 silly gunzTarPerm modified mode [ 'package.json', 436, 420 ]
40 silly gunzTarPerm extractEntry .npmignore
41 silly gunzTarPerm modified mode [ '.npmignore', 436, 420 ]
42 silly gunzTarPerm extractEntry README.md
43 silly gunzTarPerm modified mode [ 'README.md', 436, 420 ]
44 silly gunzTarPerm extractEntry LICENSE
45 silly gunzTarPerm modified mode [ 'LICENSE', 436, 420 ]
46 silly gunzTarPerm extractEntry .jscsrc
47 silly gunzTarPerm modified mode [ '.jscsrc', 436, 420 ]
48 silly gunzTarPerm extractEntry .editorconfig
49 silly gunzTarPerm modified mode [ '.editorconfig', 436, 420 ]
50 silly gunzTarPerm extractEntry .eslintrc
51 silly gunzTarPerm modified mode [ '.eslintrc', 436, 420 ]
52 silly gunzTarPerm extractEntry .eslintignore
53 silly gunzTarPerm modified mode [ '.eslintignore', 436, 420 ]
54 silly gunzTarPerm extractEntry bin/hexo
55 silly gunzTarPerm extractEntry completion/README.md
56 silly gunzTarPerm modified mode [ 'completion/README.md', 436, 420 ]
57 silly gunzTarPerm extractEntry completion/bash
58 silly gunzTarPerm modified mode [ 'completion/bash', 436, 420 ]
59 silly gunzTarPerm extractEntry completion/fish
60 silly gunzTarPerm modified mode [ 'completion/fish', 436, 420 ]
61 silly gunzTarPerm extractEntry completion/zsh
62 silly gunzTarPerm modified mode [ 'completion/zsh', 436, 420 ]
63 silly gunzTarPerm extractEntry lib/context.js
64 silly gunzTarPerm modified mode [ 'lib/context.js', 436, 420 ]
65 silly gunzTarPerm extractEntry lib/find_pkg.js
66 silly gunzTarPerm modified mode [ 'lib/find_pkg.js', 436, 420 ]
67 silly gunzTarPerm extractEntry lib/goodbye.js
68 silly gunzTarPerm modified mode [ 'lib/goodbye.js', 436, 420 ]
69 silly gunzTarPerm extractEntry lib/hexo.js
70 silly gunzTarPerm modified mode [ 'lib/hexo.js', 436, 420 ]
71 silly gunzTarPerm extractEntry lib/console/help.js
72 silly gunzTarPerm modified mode [ 'lib/console/help.js', 436, 420 ]
73 silly gunzTarPerm extractEntry lib/console/index.js
74 silly gunzTarPerm modified mode [ 'lib/console/index.js', 436, 420 ]
75 silly gunzTarPerm extractEntry lib/console/init.js
76 silly gunzTarPerm modified mode [ 'lib/console/init.js', 436, 420 ]
77 silly gunzTarPerm extractEntry lib/console/version.js
78 silly gunzTarPerm modified mode [ 'lib/console/version.js', 436, 420 ]
79 silly gunzTarPerm extractEntry lib/extend/console.js
80 silly gunzTarPerm modified mode [ 'lib/extend/console.js', 436, 420 ]
81 silly gunzTarPerm extractEntry .gitmodules
82 silly gunzTarPerm modified mode [ '.gitmodules', 436, 420 ]
83 verbose write writing to /usr/local/lib/node_modules/hexo-cli/package.json
84 info preinstall hexo-cli@1.0.1
85 verbose readDependencies loading dependencies from /usr/local/lib/node_modules/hexo-cli/package.json
86 silly prepareForInstallMany adding abbrev@^1.0.7 from hexo-cli dependencies
87 silly prepareForInstallMany adding bluebird@^3.3.0 from hexo-cli dependencies
88 silly prepareForInstallMany adding chalk@^1.1.1 from hexo-cli dependencies
89 silly prepareForInstallMany adding hexo-fs@^0.1.5 from hexo-cli dependencies
90 silly prepareForInstallMany adding hexo-log@^0.1.1 from hexo-cli dependencies
91 silly prepareForInstallMany adding hexo-util@^0.5.3 from hexo-cli dependencies
92 silly prepareForInstallMany adding minimist@^1.2.0 from hexo-cli dependencies
93 silly prepareForInstallMany adding object-assign@^4.0.1 from hexo-cli dependencies
94 silly prepareForInstallMany adding tildify@^1.1.2 from hexo-cli dependencies
95 verbose readDependencies loading dependencies from /usr/local/lib/node_modules/hexo-cli/package.json
96 silly cache add args [ 'bluebird@^3.3.0', null ]
97 verbose cache add spec bluebird@^3.3.0
98 silly cache add parsed spec Result {
98 silly cache add   raw: 'bluebird@^3.3.0',
98 silly cache add   scope: null,
98 silly cache add   name: 'bluebird',
98 silly cache add   rawSpec: '^3.3.0',
98 silly cache add   spec: '>=3.3.0 <4.0.0',
98 silly cache add   type: 'range' }
99 silly addNamed bluebird@>=3.3.0 <4.0.0
100 verbose addNamed ">=3.3.0 <4.0.0" is a valid semver range for bluebird
101 silly addNameRange { name: 'bluebird', range: '>=3.3.0 <4.0.0', hasData: false }
102 silly mapToRegistry name bluebird
103 silly mapToRegistry using default registry
104 silly mapToRegistry registry https://registry.npmjs.org/
105 silly mapToRegistry uri https://registry.npmjs.org/bluebird
106 verbose addNameRange registry:https://registry.npmjs.org/bluebird not in flight; fetching
107 silly cache add args [ 'chalk@^1.1.1', null ]
108 verbose cache add spec chalk@^1.1.1
109 silly cache add parsed spec Result {
109 silly cache add   raw: 'chalk@^1.1.1',
109 silly cache add   scope: null,
109 silly cache add   name: 'chalk',
109 silly cache add   rawSpec: '^1.1.1',
109 silly cache add   spec: '>=1.1.1 <2.0.0',
109 silly cache add   type: 'range' }
110 silly addNamed chalk@>=1.1.1 <2.0.0
111 verbose addNamed ">=1.1.1 <2.0.0" is a valid semver range for chalk
112 silly addNameRange { name: 'chalk', range: '>=1.1.1 <2.0.0', hasData: false }
113 silly mapToRegistry name chalk
114 silly mapToRegistry using default registry
115 silly mapToRegistry registry https://registry.npmjs.org/
116 silly mapToRegistry uri https://registry.npmjs.org/chalk
117 verbose addNameRange registry:https://registry.npmjs.org/chalk not in flight; fetching
118 silly cache add args [ 'hexo-fs@^0.1.5', null ]
119 verbose cache add spec hexo-fs@^0.1.5
120 silly cache add parsed spec Result {
120 silly cache add   raw: 'hexo-fs@^0.1.5',
120 silly cache add   scope: null,
120 silly cache add   name: 'hexo-fs',
120 silly cache add   rawSpec: '^0.1.5',
120 silly cache add   spec: '>=0.1.5 <0.2.0',
120 silly cache add   type: 'range' }
121 silly addNamed hexo-fs@>=0.1.5 <0.2.0
122 verbose addNamed ">=0.1.5 <0.2.0" is a valid semver range for hexo-fs
123 silly addNameRange { name: 'hexo-fs', range: '>=0.1.5 <0.2.0', hasData: false }
124 silly mapToRegistry name hexo-fs
125 silly mapToRegistry using default registry
126 silly mapToRegistry registry https://registry.npmjs.org/
127 silly mapToRegistry uri https://registry.npmjs.org/hexo-fs
128 verbose addNameRange registry:https://registry.npmjs.org/hexo-fs not in flight; fetching
129 silly cache add args [ 'hexo-log@^0.1.1', null ]
130 verbose cache add spec hexo-log@^0.1.1
131 silly cache add parsed spec Result {
131 silly cache add   raw: 'hexo-log@^0.1.1',
131 silly cache add   scope: null,
131 silly cache add   name: 'hexo-log',
131 silly cache add   rawSpec: '^0.1.1',
131 silly cache add   spec: '>=0.1.1 <0.2.0',
131 silly cache add   type: 'range' }
132 silly addNamed hexo-log@>=0.1.1 <0.2.0
133 verbose addNamed ">=0.1.1 <0.2.0" is a valid semver range for hexo-log
134 silly addNameRange { name: 'hexo-log', range: '>=0.1.1 <0.2.0', hasData: false }
135 silly mapToRegistry name hexo-log
136 silly mapToRegistry using default registry
137 silly mapToRegistry registry https://registry.npmjs.org/
138 silly mapToRegistry uri https://registry.npmjs.org/hexo-log
139 verbose addNameRange registry:https://registry.npmjs.org/hexo-log not in flight; fetching
140 silly cache add args [ 'hexo-util@^0.5.3', null ]
141 verbose cache add spec hexo-util@^0.5.3
142 silly cache add parsed spec Result {
142 silly cache add   raw: 'hexo-util@^0.5.3',
142 silly cache add   scope: null,
142 silly cache add   name: 'hexo-util',
142 silly cache add   rawSpec: '^0.5.3',
142 silly cache add   spec: '>=0.5.3 <0.6.0',
142 silly cache add   type: 'range' }
143 silly addNamed hexo-util@>=0.5.3 <0.6.0
144 verbose addNamed ">=0.5.3 <0.6.0" is a valid semver range for hexo-util
145 silly addNameRange { name: 'hexo-util', range: '>=0.5.3 <0.6.0', hasData: false }
146 silly mapToRegistry name hexo-util
147 silly mapToRegistry using default registry
148 silly mapToRegistry registry https://registry.npmjs.org/
149 silly mapToRegistry uri https://registry.npmjs.org/hexo-util
150 verbose addNameRange registry:https://registry.npmjs.org/hexo-util not in flight; fetching
151 silly cache add args [ 'minimist@^1.2.0', null ]
152 verbose cache add spec minimist@^1.2.0
153 silly cache add parsed spec Result {
153 silly cache add   raw: 'minimist@^1.2.0',
153 silly cache add   scope: null,
153 silly cache add   name: 'minimist',
153 silly cache add   rawSpec: '^1.2.0',
153 silly cache add   spec: '>=1.2.0 <2.0.0',
153 silly cache add   type: 'range' }
154 silly addNamed minimist@>=1.2.0 <2.0.0
155 verbose addNamed ">=1.2.0 <2.0.0" is a valid semver range for minimist
156 silly addNameRange { name: 'minimist', range: '>=1.2.0 <2.0.0', hasData: false }
157 silly mapToRegistry name minimist
158 silly mapToRegistry using default registry
159 silly mapToRegistry registry https://registry.npmjs.org/
160 silly mapToRegistry uri https://registry.npmjs.org/minimist
161 verbose addNameRange registry:https://registry.npmjs.org/minimist not in flight; fetching
162 silly cache add args [ 'object-assign@^4.0.1', null ]
163 verbose cache add spec object-assign@^4.0.1
164 silly cache add parsed spec Result {
164 silly cache add   raw: 'object-assign@^4.0.1',
164 silly cache add   scope: null,
164 silly cache add   name: 'object-assign',
164 silly cache add   rawSpec: '^4.0.1',
164 silly cache add   spec: '>=4.0.1 <5.0.0',
164 silly cache add   type: 'range' }
165 silly addNamed object-assign@>=4.0.1 <5.0.0
166 verbose addNamed ">=4.0.1 <5.0.0" is a valid semver range for object-assign
167 silly addNameRange { name: 'object-assign',
167 silly addNameRange   range: '>=4.0.1 <5.0.0',
167 silly addNameRange   hasData: false }
168 silly mapToRegistry name object-assign
169 silly mapToRegistry using default registry
170 silly mapToRegistry registry https://registry.npmjs.org/
171 silly mapToRegistry uri https://registry.npmjs.org/object-assign
172 verbose addNameRange registry:https://registry.npmjs.org/object-assign not in flight; fetching
173 silly cache add args [ 'tildify@^1.1.2', null ]
174 verbose cache add spec tildify@^1.1.2
175 silly cache add parsed spec Result {
175 silly cache add   raw: 'tildify@^1.1.2',
175 silly cache add   scope: null,
175 silly cache add   name: 'tildify',
175 silly cache add   rawSpec: '^1.1.2',
175 silly cache add   spec: '>=1.1.2 <2.0.0',
175 silly cache add   type: 'range' }
176 silly addNamed tildify@>=1.1.2 <2.0.0
177 verbose addNamed ">=1.1.2 <2.0.0" is a valid semver range for tildify
178 silly addNameRange { name: 'tildify', range: '>=1.1.2 <2.0.0', hasData: false }
179 silly mapToRegistry name tildify
180 silly mapToRegistry using default registry
181 silly mapToRegistry registry https://registry.npmjs.org/
182 silly mapToRegistry uri https://registry.npmjs.org/tildify
183 verbose addNameRange registry:https://registry.npmjs.org/tildify not in flight; fetching
184 silly cache add args [ 'abbrev@^1.0.7', null ]
185 verbose cache add spec abbrev@^1.0.7
186 silly cache add parsed spec Result {
186 silly cache add   raw: 'abbrev@^1.0.7',
186 silly cache add   scope: null,
186 silly cache add   name: 'abbrev',
186 silly cache add   rawSpec: '^1.0.7',
186 silly cache add   spec: '>=1.0.7 <2.0.0',
186 silly cache add   type: 'range' }
187 silly addNamed abbrev@>=1.0.7 <2.0.0
188 verbose addNamed ">=1.0.7 <2.0.0" is a valid semver range for abbrev
189 silly addNameRange { name: 'abbrev', range: '>=1.0.7 <2.0.0', hasData: false }
190 silly mapToRegistry name abbrev
191 silly mapToRegistry using default registry
192 silly mapToRegistry registry https://registry.npmjs.org/
193 silly mapToRegistry uri https://registry.npmjs.org/abbrev
194 verbose addNameRange registry:https://registry.npmjs.org/abbrev not in flight; fetching
195 verbose request uri https://registry.npmjs.org/bluebird
196 verbose request no auth needed
197 info attempt registry request try #1 at 02:10:43
198 verbose request id dd4447b98f6005f8
199 http request GET https://registry.npmjs.org/bluebird
200 verbose request uri https://registry.npmjs.org/chalk
201 verbose request no auth needed
202 info attempt registry request try #1 at 02:10:43
203 http request GET https://registry.npmjs.org/chalk
204 verbose request uri https://registry.npmjs.org/hexo-fs
205 verbose request no auth needed
206 info attempt registry request try #1 at 02:10:43
207 http request GET https://registry.npmjs.org/hexo-fs
208 verbose request uri https://registry.npmjs.org/hexo-log
209 verbose request no auth needed
210 info attempt registry request try #1 at 02:10:43
211 http request GET https://registry.npmjs.org/hexo-log
212 verbose request uri https://registry.npmjs.org/hexo-util
213 verbose request no auth needed
214 info attempt registry request try #1 at 02:10:43
215 http request GET https://registry.npmjs.org/hexo-util
216 verbose request uri https://registry.npmjs.org/minimist
217 verbose request no auth needed
218 info attempt registry request try #1 at 02:10:43
219 http request GET https://registry.npmjs.org/minimist
220 verbose request uri https://registry.npmjs.org/object-assign
221 verbose request no auth needed
222 info attempt registry request try #1 at 02:10:43
223 http request GET https://registry.npmjs.org/object-assign
224 verbose request uri https://registry.npmjs.org/tildify
225 verbose request no auth needed
226 info attempt registry request try #1 at 02:10:43
227 http request GET https://registry.npmjs.org/tildify
228 verbose request uri https://registry.npmjs.org/abbrev
229 verbose request no auth needed
230 info attempt registry request try #1 at 02:10:43
231 http request GET https://registry.npmjs.org/abbrev
232 info retry will retry, error on last attempt: Error: connect ETIMEDOUT 185.31.19.162:443
233 info retry will retry, error on last attempt: Error: connect ETIMEDOUT 185.31.19.162:443
234 info retry will retry, error on last attempt: Error: connect ETIMEDOUT 185.31.19.162:443
235 info retry will retry, error on last attempt: Error: connect ETIMEDOUT 185.31.19.162:443
236 info retry will retry, error on last attempt: Error: connect ETIMEDOUT 185.31.19.162:443
237 info retry will retry, error on last attempt: Error: connect ETIMEDOUT 185.31.19.162:443
238 info retry will retry, error on last attempt: Error: connect ETIMEDOUT 185.31.19.162:443
239 info retry will retry, error on last attempt: Error: connect ETIMEDOUT 185.31.19.162:443
240 info retry will retry, error on last attempt: Error: connect ETIMEDOUT 185.31.19.162:443
241 info attempt registry request try #2 at 02:13:00
242 http request GET https://registry.npmjs.org/abbrev
243 info attempt registry request try #2 at 02:13:00
244 http request GET https://registry.npmjs.org/bluebird
245 info attempt registry request try #2 at 02:13:00
246 http request GET https://registry.npmjs.org/hexo-fs
247 info attempt registry request try #2 at 02:13:00
248 http request GET https://registry.npmjs.org/hexo-log
249 info attempt registry request try #2 at 02:13:00
250 http request GET https://registry.npmjs.org/chalk
251 info attempt registry request try #2 at 02:13:00
252 http request GET https://registry.npmjs.org/tildify
253 info attempt registry request try #2 at 02:13:00
254 http request GET https://registry.npmjs.org/minimist
255 info attempt registry request try #2 at 02:13:00
256 http request GET https://registry.npmjs.org/hexo-util
257 info attempt registry request try #2 at 02:13:00
258 http request GET https://registry.npmjs.org/object-assign
259 info retry will retry, error on last attempt: Error: connect ETIMEDOUT 23.235.43.162:443
260 info retry will retry, error on last attempt: Error: connect ETIMEDOUT 23.235.43.162:443
261 info retry will retry, error on last attempt: Error: connect ETIMEDOUT 23.235.43.162:443
262 info retry will retry, error on last attempt: Error: connect ETIMEDOUT 23.235.43.162:443
263 info retry will retry, error on last attempt: Error: connect ETIMEDOUT 23.235.43.162:443
264 info retry will retry, error on last attempt: Error: connect ETIMEDOUT 23.235.43.162:443
265 info retry will retry, error on last attempt: Error: connect ETIMEDOUT 23.235.43.162:443
266 info retry will retry, error on last attempt: Error: connect ETIMEDOUT 23.235.43.162:443
267 info retry will retry, error on last attempt: Error: connect ETIMEDOUT 23.235.43.162:443
268 info attempt registry request try #3 at 02:16:08
269 http request GET https://registry.npmjs.org/bluebird
270 info attempt registry request try #3 at 02:16:08
271 http request GET https://registry.npmjs.org/abbrev
272 info attempt registry request try #3 at 02:16:08
273 http request GET https://registry.npmjs.org/tildify
274 info attempt registry request try #3 at 02:16:08
275 http request GET https://registry.npmjs.org/hexo-log
276 info attempt registry request try #3 at 02:16:08
277 http request GET https://registry.npmjs.org/minimist
278 info attempt registry request try #3 at 02:16:08
279 http request GET https://registry.npmjs.org/hexo-fs
280 info attempt registry request try #3 at 02:16:08
281 http request GET https://registry.npmjs.org/hexo-util
282 info attempt registry request try #3 at 02:16:08
283 http request GET https://registry.npmjs.org/object-assign
284 info attempt registry request try #3 at 02:16:08
285 http request GET https://registry.npmjs.org/chalk
286 verbose about to build /usr/local/lib/node_modules/hexo-cli
287 verbose unlock done using /home/chengca/.npm/_locks/hexo-cli-d89d95637eeaa0d8.lock for /usr/local/lib/node_modules/hexo-cli
288 verbose stack Error: connect ETIMEDOUT 185.31.18.162:443
288 verbose stack     at Object.exports._errnoException (util.js:870:11)
288 verbose stack     at exports._exceptionWithHostPort (util.js:893:20)
288 verbose stack     at TCPConnectWrap.afterConnect [as oncomplete] (net.js:1057:14)
289 verbose cwd /home/chengca/Downloads/hexo-cli-master
290 error Linux 3.13.0-55-generic
291 error argv "/usr/local/bin/node" "/usr/local/bin/npm" "install" "-g"
292 error node v4.4.0
293 error npm  v2.14.20
294 error code ETIMEDOUT
295 error errno ETIMEDOUT
296 error syscall connect
297 error network connect ETIMEDOUT 185.31.18.162:443
297 error network This is most likely not a problem with npm itself
297 error network and is related to network connectivity.
297 error network In most cases you are behind a proxy or have bad network settings.
297 error network
297 error network If you are behind a proxy, please make sure that the
297 error network 'proxy' config is set properly.  See: 'npm help config'
298 verbose exit [ 1, true ]
299 verbose unbuild lib/node_modules/hexo-cli
300 info preuninstall hexo-cli@1.0.1
301 info uninstall hexo-cli@1.0.1
302 verbose unbuild rmStuff hexo-cli@1.0.1 from /usr/local/lib/node_modules
303 silly gentlyRm /usr/local/bin/hexo is being gently removed
304 silly gentlyRm verifying /usr/local is an npm working directory
305 silly gentlyRm containing path /usr/local is under npm's control, in /usr/local
306 silly gentlyRm deletion target /usr/local/bin/hexo is under /usr/local
307 verbose gentlyRm vacuuming from /usr/local/bin/hexo up to /usr/local
308 info postuninstall hexo-cli@1.0.1
309 silly gentlyRm /usr/local/lib/node_modules/hexo-cli is being purged from base /usr/local
310 verbose gentlyRm don't care about contents; nuking /usr/local/lib/node_modules/hexo-cli
311 silly vacuum-fs purging /usr/local/lib/node_modules/hexo-cli
312 silly vacuum-fs quitting because other entries in /usr/local/lib/node_modules

**Comment by leesei** [comment]
The error is obvious from the log:
```
297 error network connect ETIMEDOUT 185.31.18.162:443
297 error network This is most likely not a problem with npm itself
297 error network and is related to network connectivity.
297 error network In most cases you are behind a proxy or have bad network settings.
```

Try using http://npm.taobao.org/ if you are behind GFW
